### ISSUE TITLE
The font / window changes after using webbrowser.open() in WSL

### ISSUE THREAD

**Issue by csm10495** [python/cpython#101838]
# Bug report

This is going to sound weird but in WSL, if you do this:

```
import webbrowser
webbrowser.open('http://google.com')
```

It doesn't really matter the URL.. for me it opens in chrome.

The font and size of the command prompt window changes.

Here is the full output:

```
csm10495@csm10495-desk:/mnt/c/Windows/System32 130 $ python3.11
Python 3.11.2 (main, Feb  8 2023, 14:49:24) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import webbrowser
>>> webbrowser.open('http://google.com')
mkdir: cannot create directory '/home/csm10495/.config/wslu': Permission denied
/usr/bin/www-browser: line 137: /home/csm10495/.config/wslu/oemcp: No such file or directory
cat: /home/csm10495/.config/wslu/oemcp: No such file or directory
/usr/bin/www-browser: line 119: /home/csm10495/.config/wslu/baseexec: No such file or directory
cat: /home/csm10495/.config/wslu/oemcp: No such file or directory
True
>>>
```

Here is a gif since it sounds really odd and tough to fully explain: 

![weird](https://user-images.githubusercontent.com/5749838/218293640-1ca3c749-a5a0-46a4-b629-e1aca4287839.gif)

# Your environment

WSL (Ubuntu 20.04.5 LTS) on Windows 10 x64 (10.0.19045.2364])
```
csm10495@csm10495-desk:/mnt/c/Windows/System32 $ python3.11 --version
Python 3.11.2
```

```
csm10495@csm10495-desk:/mnt/c/Windows/System32 $ uname -a
Linux csm10495-desk 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by csm10495** [comment]
This must be something specific to do with wsl weirdness. I traced it down to the subprocess call to `www-browser`.. that seems to be similar to calling the wsl utility `wslview`. Same behavior happens if you call: `wslview http://google.com`. 

With that, i filed an issue against the Ubuntu version of the `wslu` package: https://bugs.launchpad.net/ubuntu/+source/wslu/+bug/2007017.

TLDR: It's not Python's fault. :) Closing this out.

**Comment by eryksun** [comment]
> www-browser.. that seems to be similar to calling the wsl utility wslview

Yes, "/usr/bin/www-browser" should be a symlink to "/etc/alternatives/www-browser", which should be a symlink to the shell script "/usr/bin/wslview". 

For some reason, you don't have permission to create the ".config/wslu" directory in your home directory, and it doesn't exist. It should exist. Not being able to read the "oemcp" (OEM code page) file from that directory is probably the cause of the problem.

The wslview script contains a function that runs the system version of Windows PowerShell 5. This function begins by trying to change the console code page from 65001 (UTF-8), as used by WSL, to the OEM code page.

```sh
function chcp_com {
        "$(interop_prefix)"c/Windows/System32/chcp.com "$@" >/dev/null
}

function winps_exec {
        chcp_com $(cat ~/.config/wslu/oemcp)
        "$(interop_prefix)"c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -NoProfile -NonInteractive -Command "$@"
        chcp_com 65001
}
```

Since the "oemcp" file doesn't exist, the "chcp.com" command is passed no arguments and simply reports the current code page (redirected to "/dev/null") instead of changing it. 

If the current console code page is 65001, depending on the current font, then running powershell.exe switches the screen buffer of a classic console session to "Raster Fonts". (This does not affect a pseudoconsole session, such as Windows Terminal.) I observe this when the current console font is Consolas, but not if it's Cascadia Code, Courier New, or Lucida Console.
